### ISSUE TITLE
Support Serverless >=1.60.2

### ISSUE THREAD

**Issue by jaikdean** [dougmoscrop/serverless-plugin-split-stacks#106]
There was a change introduced in Serverless 1.60.2 that is causing issues with this plugin. See [this issue](https://github.com/serverless/serverless/issues/7206) on the Serverless project.

I'd love to help but have no idea where to start. [This comment](https://github.com/serverless/serverless/issues/7206#issuecomment-575569367) seems to outline the issue.

**Comment by dougmoscrop** [comment]
Yep, so much fun. I wish they had an API for manipulating things instead of raw. I think I fixed this in another plugin so I will have a look.

**Comment by aaronwhite** [comment]
New comer - we've run into this as well; @dougmoscrop if you can point me at the fix in your other plugin, happy to take a stab at replicating here

**Comment by dougmoscrop** [comment]
My bad, I think I confused my serverless bugs that I fixed, or just mixed up "mental note: todo" with "did". lemme see what I can do tonight.

**Comment by dougmoscrop** [comment]
but seriously thank you for offering to help instead of just asking when it will be fixed :)

**Comment by dougmoscrop** [comment]
Hmm, I just tested with serverless 1.63 and 1.64.1 and it worked in my test service. That doesn't mean it will work in all cases. I tried 1.60.2 and it also worked.

Are you sure it's this plugin? What other plugins are you using?

**Comment by jelder** [comment]
Hi @dougmoscrop, I work with @aaronwhite. In our specific case, we're using split-stacks in conjunction with layers. We partition our lambdas into stacks based on the first letter of the logical name, and each stack that uses a layer gets an error similar to the above.

Here's our split-stacks config: https://gist.github.com/jelder/6336ae8209fab7f6fd3b70d4446b1942

The error we get on serverless 1.65.0: 

```
An error occurred: LambdaMNestedStack - Template format error:
Unresolved resource dependencies [NodeModulesLambdaLayerParameter]
in the Resources block of the template.
```

Inspecting the files in `.serverless/cloudformation-template-*-nested-stackjson`, each lambda has this (as expected):

```json
"Layers": [
  {
    "Ref": "NodeModulesLambdaLayerParameter"
  }
],
```

But the nested stacks lack a matching value in the `Parameters` section. This seemed similar to the problem outlined in https://github.com/serverless/serverless/issues/7206, but apologies if we're off the mark there.

**Comment by dougmoscrop** [comment]
Are the layers generated using a plugin or anything? Can you share as much of your serverless config as possible (you can email it to me if you don't want to make it public)

**Comment by jormaechea** [comment]
I had the same issue with a simple configuration: two functions in different nested stacks (using stacks-map.js), and a security group in the base stack.
Then I set the vpc config under `provider` object.
I can give you a sample config file tomorrow!

**Comment by jelder** [comment]
No plugins, just a predeploy script we wrote that prepares a zip of node_modules. I'll try to produce a minimal reproduction to share.

**Comment by jormaechea** [comment]
@dougmoscrop Well, it took me a little longer, but here it is. The most simple service I could create to demonstrate the error: https://gist.github.com/jormaechea/19ee0642f3eaeaba1f8c3fcefd02ea1c

It's just a security group and 2 lambda functions. Each lambda function in a separate nested stack.

If you run `sls package` you will see that one nested stack receives the `MySGParameter`, but the other doesn't (although it tries to use it).

Hope it helps to narrow down the search!

**Comment by dougmoscrop** [comment]
Thanks! I will take a look

**Comment by jormaechea** [comment]
@dougmoscrop any idea where the issue may be so we can contribute with a PR? I looked around but don't know where I should be paying attention to :stuck_out_tongue:

**Comment by jormaechea** [comment]
My best guess is this https://github.com/dougmoscrop/serverless-plugin-split-stacks/blob/master/lib/replace-references.js but maybe you can point to some specific part of the file..

**Comment by dougmoscrop** [comment]
Yeah that's a good suspect, but I just can't see (with what little spare time I have right now) how it would go wrong

**Comment by jormaechea** [comment]
Well, it's been a couple hours, but I think i found it! I had to (fix and) run the integration tests with my sample `serverless.yml` and everything points to some unwanted side effect.

The problem is that when you replace some parameter it gets changed, so the second time it tries to be replaced it has already changed and chaos happens.

I'll test it a little more, clean up the code and then submit a PR.. :tada:

**Comment by gevorggalstyan** [comment]
@jormaechea how did your testing go? Did you find a solution? Would you mind sharing it?

**Comment by luanmuniz** [comment]
Just to ping this topic. I'm also having the Parameter problem.

**Comment by jormaechea** [comment]
@luanmuniz can you try installing the plugin using the git url, and confirm if it's solved?

´npm i https://github.com/dougmoscrop/serverless-plugin-split-stacks´

Thanks!

**Comment by luanmuniz** [comment]
@jormaechea The Parameter Problem seems to be solved, but then I got another error:

```
Error: The CloudFormation template is invalid: Template error: instance of Fn::GetAtt references undefined resource LoginWithPasswordLambdaFunctionRole
```

Just so you know I have the perType config active with some stack-map custom configs.

**Comment by jormaechea** [comment]
@luanmuniz I think I got it right now!! Can you try again, installing from my fork's branch from PR #111 ?

**Comment by luanmuniz** [comment]
@jormaechea The deployment worked, no problems with the template. I will test the actual deployed code and compare the 1.59 generated files with your version tomorrow and get back to you.

Sorry it took me so long, the problems only happen when deploying to production, so I need to be extra careful.

**Comment by jormaechea** [comment]
Luan, I found some errors in the PR. Please do not test this in a production environment...

**Comment by luanmuniz** [comment]
@jormaechea Thanks for the notice. Don't worry about the production environment, with everything that's happening in the world, the production environment is not that sensible right now. But I will be extra-extra-careful from now on!

Let me know when I can check again.

**Comment by andy265** [comment]
After updating from serverless 1.33 to the latest got this problem for `subnets` and `securityGroups` in `vpc`, and for `ApiGatewayRestApi`.
